src/Library/demos/Web View/main.js

```diff
@@ -0,0 +1,91 @@
+import WebKit from "gi://WebKit?version=6.0";
+const GObject = imports.gi.GObject;
+const { Uri } = imports.gi.GLib;
+
+const container = workbench.builder.get_object("container");
+const button_back = workbench.builder.get_object("button_back");
+const button_forward = workbench.builder.get_object("button_forward");
+const button_reload = workbench.builder.get_object("button_reload");
+const button_stop = workbench.builder.get_object("button_stop");
+const url_bar = workbench.builder.get_object("url_bar");
+const web_view = new WebKit.WebView({
+  zoom_level: 0.8,
+});
+container.child = web_view;
+
+// URL bar displays the current loaded page
+web_view.bind_property(
+  "uri",
+  url_bar.buffer,
+  "text",
+  GObject.BindingFlags.DEFAULT,
+);
+
+// URL bar displays the current loaded page
+web_view.load_uri("https://www.gnome.org/");
+
+url_bar.connect("activate", () => {
+  let url = url_bar.buffer.text;
+  const scheme = Uri.peek_scheme(url);
+  if (scheme == null) {
+    url = `https://${url}`;
+  }
+  web_view.load_uri(url);
+});
+
+button_forward.connect("clicked", () => {
+  web_view.go_forward();
+});
+
+button_back.connect("clicked", () => {
+  web_view.go_back();
+});
+
+button_reload.connect("clicked", () => {
+  web_view.reload();
+});
+
+button_stop.connect("clicked", () => {
+  web_view.stop_loading();
+});
+
+web_view.connect("load-changed", (view, load_event) => {
+  switch (load_event) {
+    case WebKit.LoadEvent["STARTED"]:
+      console.log("Page loading started");
+      break;
+    case WebKit.LoadEvent["FINISHED"]:
+      console.log("Page loading has finished ");
+      break;
+  }
+});
+
+web_view.connect("load-failed", (view, load_event, fail_url, error) => {
+  // Dont display error page if it is caused by stop_loading()
+  if (error.code !== 302) {
```

Comment:
any idea i there is an enum we can use instead?

Reply:
Yeah I just found ``WebKit.NetworkError.CANCELLED``, I can use that instead.

Reply:
yes please, it's always good to avoid "magic numbers"

https://en.wikipedia.org/wiki/Magic_number_(programming)